src/cli.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright 2020 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+
+//! CLI implementation.
+
+use crate::common::{PROJECT_AUTHOR, PROJECT_DESC, PROJECT_NAME, PROJECT_VERSION};
+use clap::{App, AppSettings, Arg};
+
+#[derive(Debug)]
+pub enum Action {
+    ListOpcodes(String),
+    ListProviders,
+    Ping,
+}
+
+#[derive(Debug)]
+pub struct CommandLineContext {
+    pub action: Action,
+    pub app_name: Option<String>,
+    pub verbosity: u64,
+}
+
+pub fn parse_cli() -> CommandLineContext {
+    let matches = App::new(PROJECT_NAME)
+        .about(PROJECT_DESC)
+        .author(PROJECT_AUTHOR)
+        .version(PROJECT_VERSION)
+        .setting(AppSettings::SubcommandRequiredElseHelp)
+        .arg(
+            Arg::with_name("v")
+                .short('v')
+                .multiple(true)
+                .about("Sets the level of verbosity"),
+        )
+        .subcommand(
+            App::new("list_opcodes").about("List the opcodes").arg(
+                Arg::with_name("provider")
+                    .short('p')
+                    .takes_value(true)
+                    .required(true)
```

Comment:
Could the `p` argument be not required and if not passed, the `list_opcodes` command would execute on the default provider (the first one returned by `list_providers`)?

Reply:
Good idea, will change this.